standard/classes.md

```diff
@@ -5542,3 +5544,129 @@ When the body of the async function terminates, the return task is moved out of
+- The type does not have an accessible instance indexer taking a `System.Index` as its only argument, or as its first argument with the remaining arguments being optional.
+
+The provided instance indexer shall have the same get and set members with matching accessibility as the `int` indexer.
+
+The provided instance indexer shall take the given `System.Index` and use that to call the instance indexer taking an `int`. If both the `Length` and `Count` properties exist and are accessible, `Length` is used.
+
+> *Note*: See §indexable-sequence-expl-support-for-index for an example of an explicitly provided `Index` indexer. If that were not defined, its equivalent would be provided by the implementation. *end note*
+
+### §indexable-sequence-support-for-range Providing support for Range
+
+#### §indexable-sequence-expl-support-for-range Explicit Range support
+
+A type having an instance indexer taking a single argument of type `System.Range`, or a first argument of that type followed by optional arguments, may be indexed as described by [§12.8.11.3](expressions.md#128113-indexer-access).
+
+> *Example*: In [§15.9](classes.md#159-indexers), there is an example defining type `BitArray`, which stores bits in an array of `int`. Adding a `Range` indexer that returns a `BitArray` representing the bit slice designated by the Range, is simple:
+>
+> <!-- Example: {template:"standalone-console", name:"ExplicitRangeSupport", additionalFiles:["BitArrayPartial2.cs"], expectedOutput:["ba = >10011<","BitArray is >10011<","BitArray is >10011<","BitArray is >1<","BitArray is >1<","BitArray is ><"]} -->
+> ```csharp
+> partial class BitArray
+> {
+>     public BitArray this[Range range]       // note the return type
+>     {
+>         get
+>         {
+>             int startIdx = range.Start.GetOffset(Length);
+>             int endIdx = range.End.GetOffset(Length);
+>             int rangeLength = endIdx - startIdx;
+>             BitArray newBitArray = new BitArray(rangeLength);
+>             for (int i = startIdx; i < endIdx; ++i)
+>             {
+>                 newBitArray[i - startIdx] = this[i];
+>             }
+>             return newBitArray;
+>         }
+>     }
+> }
+> ```
+>
+> *end example*
+
+#### §indexable-sequence-impl-support-for-range Implicit Range support
+
+An implementation shall behave as if it provides a non-virtual instance indexer member with a single parameter of type `System.Range` for any type that meets the following criteria:
+
+- The type is countable ([§15.7.1](classes.md#1571-general)).
+- The type has an accessible instance method named `Slice` taking two arguments of type `int` as the only arguments. For type `string`, the method `Substring` is used instead of `Slice`.
+  > *Note*: As specified in [§12.8.11.2](expressions.md#128112-array-access), for array access, the method `System.Runtime.CompilerServices.RuntimeHelpers.GetSubArray` is used instead of `Slice`. *end note*
```

Comment:
section change:

```suggestion
  > *Note*: As specified in [§12.8.12.2](expressions.md#128122-array-access), for array access, the method `System.Runtime.CompilerServices.RuntimeHelpers.GetSubArray` is used instead of `Slice`. *end note*
```